erpnext/hr/report/employee_leave_balance/employee_leave_balance.py

```diff
@@ -52,11 +54,13 @@ def get_data(filters, leave_types):
 
 	active_employees = frappe.get_all("Employee",
 		filters=conditions,
-		fields=["name", "employee_name", "department", "user_id"])
+		fields=["name", "employee_name", "department", "user_id", "leave_approver"])
+
+	department_approver_map = get_department_leave_approver_map(filters.get('department'))
 
 	data = []
 	for employee in active_employees:
-		leave_approvers = get_approvers(employee.department)
+		leave_approvers = employee.leave_approver or department_approver_map.get(employee.department_name, [])
```

Comment:
Leave Approver mentioned in Employee is default approver. All the other approvers for the department should also be able to view the report.

You should just add employee.leave_approver to department_leave_approvers list, if not exists.